chia/data_layer/data_store.py

```diff
@@ -1460,7 +1480,11 @@ async def insert_batch(
                             if key_hash_frequency[hash] == 1 or (
                                 key_hash_frequency[hash] == 2 and first_action[hash] == "delete"
                             ):
-                                old_node = await self.maybe_get_node_by_key(key, store_id)
+                                old_node: Optional[Node] = None
+                                if hash in leaf_hashes:
+                                    leaf_hash = leaf_hashes[hash]
+                                    old_node = await self.get_node(leaf_hash)
+                                    assert isinstance(old_node, TerminalNode)
```

Comment:
Instead of inducing the duplication of this, maybe either `.maybe_get_node_by_key()` could do this if passed the leaf hashes, or a new method for that?